force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,49 @@
+trigger OpportunityTrigger on Opportunity (before delete, before update) {
+    if (Trigger.isBefore && Trigger.isDelete) {
+        // Question 6 solution               
+        Set<Id> accIds = new Set<Id>();
+        for (Opportunity opp : Trigger.old) {
+            if (opp.AccountId != null) {
+                accIds.add(opp.AccountId);
+            }
```

Comment:
this is good. A different approach would be to add the ids where the Opp = closed won. So you only query for those.